### ISSUE TITLE
Problem with Inflector

### ISSUE THREAD

**Issue by ElBeni1980** [cakephp/cakephp#6659]
Since 2.6.5 I have problems con multi-word irregular nouns (with 3 or more words). In 2.6.4 this works fine.

Example:

I define (in spanish) with the Inflector::rules the plural "categorias_preguntas_frecuentes" for the singular "categoria_pregunta_frecuente"... But in the model the application looks for the table "faq_categoria_preguntas_frecuentes" instead the correct table "faq_categorias_preguntas_frecuentes". My model is into a plugin named Faq (tables start with "faq_").

I have obseved that now I need to prepend the name of the plugin into the expressions like "faq_categorias_preguntas_frecuentes" for the plural and "faq_categoria_pregunta_frecuente" for the singular in order to keep the application working... In previous versions this was not required. ¿Is this behaviour correct or is an error...?

Thank you so much for your attention.

**Comment by markstory** [comment]
How are you defining the custom inflection rules? There was a change in 2.6.5 to fix #6538. Knowing how to reproduce the issue you're having will help fix both :smile:

**Comment by ElBeni1980** [comment]
In bootstrap.php I have defined:

    $plural_to_singular = array(
	    'categorias_preguntas_frecuentes' => 'categoria_pregunta_frecuente'
    );

    Inflector::rules('singular', array('rules' => array(), 'irregular' => $plural_to_singular, 'uninflected' => array()));

    Inflector::rules('plural', array('rules' => array(), 'irregular' => array_flip($plural_to_singular), 'uninflected' => array()));

Then, when I'm trying to load de model CategoriaPreguntaFrecuente in the "Faq" plugin it throws:

    Missing Database Table

    Error: Table faq_categoria_preguntas_frecuentes for model CategoriaPreguntaFrecuente was not found in datasource desarrollo.

I have tested the pluralize function in bootstrap.php and works fine

    $plural_to_singular = array(
	    'categorias_preguntas_frecuentes' => 'categoria_pregunta_frecuente'
    );

    Inflector::rules('singular', array('rules' => array(), 'irregular' => $plural_to_singular, 'uninflected' => array()));

    Inflector::rules('plural', array('rules' => array(), 'irregular' => array_flip($plural_to_singular), 'uninflected' => array()));

    debug(Inflector::pluralize('categoria_pregunta_frecuente'));die;

This code produces:

    \app\Config\bootstrap.php (line 74)

    'categorias_preguntas_frecuentes'

**Comment by markstory** [comment]
I might be doing something wrong but:

```php
$inflectorMap = [
	'categorias_preguntas_frecuentes' => 'categoria_pregunta_frecuente'
];

Inflector::rules('singular', [
	'irregular' => $inflectorMap
]);
Inflector::rules('plural', [
	'irregular' => array_flip($inflectorMap)
]);

debug(Inflector::pluralize('categoria_pregunta_frecuente'));
debug(Inflector::pluralize('faq_categoria_pregunta_frecuente'));
```

outputs:

```
categorias_preguntas_frecuentes
faq_categorias_preguntas_frecuentes
```

Which from what I think you are looking for. I ran this against the head of `2.6`.

**Comment by ElBeni1980** [comment]
Ok, I think I found where is the error. I cut my code in previous examples in order to show a short example... but I think cut more... The next example (in ```bootstrap.php```) shows what it's happening:

    $plural_to_singular = array(
        'categorias_preguntas_frecuentes' => 'categoria_pregunta_frecuente',
        'preguntas_frecuentes' => 'pregunta_frecuente',
    );

    Inflector::rules('singular', array('rules' => array(), 'irregular' => $plural_to_singular, 'uninflected' => array()));

    Inflector::rules('plural', array('rules' => array(), 'irregular' => array_flip($plural_to_singular), 'uninflected' => array()));

    debug(Inflector::pluralize('categoria_pregunta_frecuente'));
    debug(Inflector::pluralize('faq_categoria_pregunta_frecuente'));

This code produces:

    'categoria_preguntas_frecuentes'
    'faq_categoria_preguntas_frecuentes'

The second rule I have defined causes ```Inflector``` to fail. I suppose this behavior is because ```'preguntas_frecuentes'``` is contained in ```'categorias_preguntas_frecuentes'``` (and ```'pregunta_frecuente'``` is contained in ```'categoria_pregunta_frecuente'```).

**Comment by markstory** [comment]
Ah ok, that is good to know. You mentioned earlier that the Inflector calls in bootstrap.php worked fine. Is it just the model to table name inflection that is failing?

**Comment by ElBeni1980** [comment]
I mentioned earlier that the Inflector calls in bootstrap.php worked fine, but I was wrong. My examples were based in the code I cut... and they not show the code real where I found the bug.

In my previous comment I show the code making ```Inflector``` fail in the ```bootstrap.php```. My previous comment is what should really serve as example to show how the ```Inflector``` fails.

**Comment by markstory** [comment]
Ok, I think I've figured out the problem finally. The 'preguntas_frecuentes' inflection is being applied after the 'categorias_preguntas_frecuentes' inflection and corrupting the result.

**Comment by markstory** [comment]
Fixed with #6723
